README.md

```diff
@@ -1 +1,253 @@
+      * Handle pre-deepsleep operations if any are necessary, like manually disabling
+      * oscillators, change clock settings, etc.
+      ******************************************************************************/
+      void sli_power_manager_handle_pre_deepsleep_operations(void)
+        {
+          if (is_hf_x_oscillator_used) {
+              CMU_OscillatorEnable(cmuOsc_HFRCO, true, true);
+              CMU_HFRCOBandSet(cmuHFRCOFreq_38M0Hz);
+              CMU_ClockSelectSet(cmuClock_HF, cmuSelect_HFRCO);
+              CMU_OscillatorEnable(cmuOsc_HFXO, false, false);
+              sl_board_disable_oscillator(SL_BOARD_OSCILLATOR_TCXO);
+          }
+        } 
+        
+        /***************************************************************************//**
+      * Handle post-sleep operations if any are necessary, like manually enabling
+      * oscillators, change clock settings, etc.
+      ******************************************************************************/
+      void sli_power_manager_restore_high_freq_accuracy_clk(void)
+      {
+        if (is_hf_x_oscillator_used) {
+            sl_board_enable_oscillator(SL_BOARD_OSCILLATOR_TCXO);
+        }
+      } 
+      ```
+
+   -  Include The TCXO control head file. Add the ***#include "sl_board_control.h"*** to this file. 
+   -  Add the ***sli_power_manager_private.h*** path to compiler. **Right click the project**->**properities**->**C/C++ build**->**setting**->**Tool setting**->**GNU ARM compiler**->**includes**, add the ***"${StudioSdkPath}/platform/service/power_manager/src"*** to path as the following figure.  
+      <img src="images/include path.png">
+
+8. **Calculate the minimum of Rx on time**
+   - Preamble detect need at least 40 bits. The formula is as follows.  
+***Trx = 40 / 9.6 = 4.167 ms***
+   - Find ***the sl_duty_cycle_config.h*** in **config** catalog, and Modify the macro **DUTY_CYCLE_ON_TIME** to 4167.
+      ```C
+      #define DUTY_CYCLE_ON_TIME      (4167)
+      ```   
+9. **Enable EM2 mode and disable button**  
+Find the ***the sl_duty_cycle_config.h*** file as above, and enable EM2 mode and disable button as following.
+    ```c
+    #define DUTY_CYCLE_USE_LCD_BUTTON      0
+    #define DUTY_CYCLE_ALLOW_EM2           1
+    ```
+10. **Build project and flash to device**
+    - Click the hammer icon build the current project.
+    - After compiling is completed, find the binary file is located in **binaries** catalog, right click->**Flash to device...**->**choose the target device**->**OK**.    
+
+### Test the sleep current
+-  Build the project and flash the hex file to target board.
+-  Use the instrument to test sleep current. The sleep current as the following figure.  
+    <img src="images/sleep current.gif">  
+
+- Conclusion  
+The sleep current is **1.94uA** after disable TCXO and unused GPIO. Due to there still some GPIO is working, so the result is in accord with the data from datasheet.  
```

Comment:
According to the EFR32FG14 datasheet, the above EM2 current consumption is close to the value mentioned in the datasheet. There still several peripherals are in use so the test result is expected.